### DISCUSSION TITLE
TextBlock and resetting _textLayout in Arrange and Measure

### DISCUSSION THREAD

**Discussion by msneijders** [AvaloniaUI/Avalonia#9826] [Q&A]
I profiled a DataGrid (with only TextBlocks in cells) and a simple ItemsRepeater with 3 TextBlocks per item.

A big part of cpu time is, not surprisingly, in `TextBlock.CreateTextLayout(string? text)`. What did surprise me was it looked like it was doing this two times for each TextBlock that needed painting (while scrolling).

Question: does anyone know the reason this textlayout cache is reset in:
- https://github.com/AvaloniaUI/Avalonia/blob/740f6afffc6a5b8cd2bc4113081b0285ac3844a0/src/Avalonia.Controls/TextBlock.cs#L729
- https://github.com/AvaloniaUI/Avalonia/blob/740f6afffc6a5b8cd2bc4113081b0285ac3844a0/src/Avalonia.Controls/TextBlock.cs#L666

I removed those lines, and got about 29% increase in frames per second in my tests. All avaloniaui unit tests still run. But those resets probably exists for a reason?

**Comment by timunie** [comment]
DataGrid is a complex Control tbh. Probably worth to send a PR, but I can't gurantee it will be merged. Anyway, thanks for your findings and sharing it with us.

**Comment by maxkatz6** [comment]
CC @Gillibald
